packages/http/test/multipart.test.ts

```diff
@@ -153,6 +153,56 @@ describe("resolving part payload", () => {
     strictEqual(body.parts[0].headers[0].options.name, "part-header");
   });
 
+  describe("HttpFile part", () => {
+    it("emit diagnostic if adding extra properties to File", async () => {
+      const diagnostics = await diagnoseHttpOp(`
+        model InvalidFile extends File {
+          @header extra: string;
+        }
+        `);
+
+      expectDiagnostics(diagnostics, {
+        code: "@typespec/http/http-file-extra-property",
+        message: "File model cannot define extra properties. Found 'extra'.",
+      });
+    });
+
+    it("use of HttpFile resolve optional filename", async () => {
+      const op = await getHttpOp(`
+        op read(
+          @header contentType: "multipart/mixed",
+          @multipartBody body: [
+            HttpPart<File, #{ name: "file" }>
+          ]): void;
+        `);
+
+      const body = op.parameters.body;
+      strictEqual(body?.bodyKind, "multipart");
+      strictEqual(body.parts.length, 1);
+      ok(body.parts[0].filename, "Filename property should have been resolved");
+      strictEqual(body.parts[0].filename.optional, true);
+    });
+
+    it("use of custom HttpFile with required filename resolve required", async () => {
+      const op = await getHttpOp(`
+        model FileWithFilename extends File {
+          filename: string;
+        }
+        op read(
+          @header contentType: "multipart/mixed",
+          @multipartBody body: [
+            HttpPart<FileWithFilename, #{ name: "file" }>
+          ]): void;
+        `);
+
+      const body = op.parameters.body;
+      strictEqual(body?.bodyKind, "multipart");
+      strictEqual(body.parts.length, 1);
+      ok(body.parts[0].filename, "Filename property should have been resolved");
```

Comment:
With this PR, author could set filename as required or not explicitly. What about "content-type"? If author also wants to set `content-type` of this httppart as required, current API can't show the info.

Reply:
Content type is already exposed in the body as content type

Reply:
File name is the only special datat that’s not part of the body

Reply:
According to https://datatracker.ietf.org/doc/html/rfc7578#section-4.4, content-type is not `MUST` but `MAY` or `SHOULD` which means that it is not necessary part in payload. However, now Typespec author could use `httpFile` to mark `filename` and `content-type` as required, so Typespce compiler shall expose the requiredness info for language emitters. Currently https://github.com/microsoft/typespec/blob/e1bf4607ede6e0de4896607e80440b70e3363e0b/packages/http/src/types.ts#L445-L448, contentTypes of body is `string[]` and may have default value set by Typespec compiler which language emitter can't infer the requiredness for it.
nit: Java/.NET may need it so add @chunyu3 and @weidongxu-msft
![image](https://github.com/microsoft/typespec/assets/70930885/67fa3909-836c-455f-b371-63a68f12b3ea)

Reply:
There is `headers`? Does `content-type` header belongs to it?
https://github.com/microsoft/typespec/blob/e1bf4607ede6e0de4896607e80440b70e3363e0b/packages/http/src/types.ts#L493-L494

Reply:
> There is `headers` in `HttpOperationPart`? Does `content-type` header belongs to it?
> 
> https://github.com/microsoft/typespec/blob/e1bf4607ede6e0de4896607e80440b70e3363e0b/packages/http/src/types.ts#L493-L494

`headers` here contains headers except `content-type`

Reply:
For meta-data like `filename` and `content-type` of each part, if customer define this those meta-data in spec, .NET will expose this meta-data as parameter or property to provide customer the capability to set. If not, .NET will not expose it. So for .net, we need to know if the `content-type` is defined or not as `filename`

Reply:
Added `contentTypeProperty` to the `HttpOperationBody`